From e5b7619b61acb9886e5600155b5d0d5e14fa9a31 Mon Sep 17 00:00:00 2001
From: LPFchan <me@lost.plus>
Date: Thu, 7 May 2026 12:15:36 +0900
Subject: [PATCH 8/9] auth: request upload bearer tokens

---
 backend/api.go | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/backend/api.go b/backend/api.go
index dad82cd..22f6e89 100644
--- a/backend/api.go
+++ b/backend/api.go
@@ -115,21 +115,13 @@ func (a *Api) getAuthToken() (map[string]string, error) {
 		return nil, fmt.Errorf("failed to parse auth data: %w", err)
 	}
 
-	authRequestData := url.Values{
-		"androidId":                    {authDataValues.Get("androidId")},
-		"app":                          {"com.google.android.apps.photos"},
-		"client_sig":                   {authDataValues.Get("client_sig")},
-		"callerPkg":                    {"com.google.android.apps.photos"},
-		"callerSig":                    {authDataValues.Get("callerSig")},
-		"device_country":               {authDataValues.Get("device_country")},
-		"Email":                        {authDataValues.Get("Email")},
-		"google_play_services_version": {authDataValues.Get("google_play_services_version")},
-		"lang":                         {authDataValues.Get("lang")},
-		"oauth2_foreground":            {authDataValues.Get("oauth2_foreground")},
-		"sdk_version":                  {authDataValues.Get("sdk_version")},
-		"service":                      {authDataValues.Get("service")},
-		"Token":                        {authDataValues.Get("Token")},
+	authRequestData := url.Values{}
+	for k, v := range authDataValues {
+		authRequestData[k] = v
 	}
+	authRequestData.Set("app", "com.google.android.apps.photos")
+	authRequestData.Set("callerPkg", "com.google.android.apps.photos")
+	authRequestData.Del("it_caveat_types")
 
 	headers := map[string]string{
 		"Accept-Encoding": "gzip",
